File: android/app/src/main/kotlin/com/example/flutter_lab/usecase/location/GetLocationUseCase.kt
package com.example.flutter_lab.usecase.location

import com.example.flutter_lab.domain.location.entity.Location
import com.example.flutter_lab.domain.location.repository.LocationRepository

class GetLocationUseCase(
    private val locationRepository: LocationRepository
) {
    suspend operator fun invoke(): Location {
        // if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
        //     ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 1)
        //     return ""
        // }
        return locationRepository.get()
    }
}

File: lib/domain/use_cases/location/get_location_use_case.dart
import 'package:flutter_lab/domain/location/location.dart';
import 'package:flutter_lab/domain/location/location_repository.dart';
import 'package:result_dart/result_dart.dart';

class GetLocationUseCase {
  GetLocationUseCase({
    required LocationRepository locationRepository,
  }) : _locationRepository = locationRepository;

  final LocationRepository _locationRepository;

  AsyncResult<Location> call() async => _locationRepository.get();
}
